src/main/java/com/amazon/sqs/javamessaging/ExtendedClientConfiguration.java

```diff
@@ -214,4 +224,72 @@ public ExtendedClientConfiguration withAlwaysThroughS3(boolean alwaysThroughS3)
 	public boolean isAlwaysThroughS3() {
 		return alwaysThroughS3;
 	}
+
+	/**
+	 * Sets whether or not messages are deleted in S3 when they are delete from
+	 * the queue.
+	 *
+	 * @param retainS3Messages
+	 *            Whether or not messages are deleted in S3 when they are delete
```

Comment:
The method name is `retain` but the description is `delete` :)

Reply:
Well, that is because this method controls if the S3 objects are either 'retained' or 'deleted', so yes the description uses the word 'deleted'. Not sure what the issue is tbh? 🤔